examples/regression/2-advanced-analysis/plot_conformal_predictive_system.py

```diff
@@ -0,0 +1,138 @@
+)
+X_train, X_test, y_train, y_test = train_test_split(
+    X, y, test_size=0.5, random_state=random_state
+)
+
+plt.xlabel("x")
+plt.ylabel("y")
+plt.scatter(X_train, y_train, alpha=0.3)
+plt.show()
+
+
+##############################################################################
+# 2. Defining a Conformal Predictive Distribution class with MAPIE
+# ----------------------------------------------------------
+#
+# To be able to obtain the cumulative distribution function of
+# a prediction with MAPIE, we propose here to wrap the
+# :class:`~mapie.regression.MapieRegressor` to add a new method named
+# `get_cumulative_distribution_function`.
+
+class MapieConformalPredictiveDistribution(MapieRegressor):
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.conformity_score.sym = False
+
+    def get_cumulative_distribution_function(self, X):
+        y_pred = self.predict(X)
+        cs = self.conformity_scores_[~np.isnan(self.conformity_scores_)]
+        res = self.conformity_score_function_.get_estimation_distribution(
+            X, y_pred.reshape((-1, 1)), cs
+        )
+        return res
+
+
+##############################################################################
+# Now, we propose to use it with two different conformity scores -
+# :class:`~mapie.conformity_score.AbsoluteConformityScore` and
+# :class:`~mapie.conformity_score.ResidualNormalisedScore` - in split-conformal
+# inference.
+
+mapie_regressor_1 = MapieConformalPredictiveDistribution(
+    estimator=LinearRegression(),
+    conformity_score=AbsoluteConformityScore(),
+    cv='split',
+    random_state=random_state
+)
+
+mapie_regressor_1.fit(X_train, y_train)
+y_pred_1 = mapie_regressor_1.predict(X_test)
+y_cdf_1 = mapie_regressor_1.get_cumulative_distribution_function(X_test)
+
+mapie_regressor_2 = MapieConformalPredictiveDistribution(
+    estimator=LinearRegression(),
+    conformity_score=ResidualNormalisedScore(),
+    cv='split',
+    random_state=random_state
+)
+
+mapie_regressor_2.fit(X_train, y_train)
+y_pred_2 = mapie_regressor_2.predict(X_test)
+y_cdf_2 = mapie_regressor_2.get_cumulative_distribution_function(X_test)
+
+plt.xlabel("x")
+plt.ylabel("y")
+plt.scatter(X_test, y_test, alpha=0.3)
+plt.plot(X_test, y_pred_1, color="C1")
+plt.show()
+
+
+##############################################################################
+# 3. Visualizing the cumulative distribution function
+# ---------------------------------------------------
+#
+# We now propose to visualize the cumulative distribution functions of
+# the predictive distribution in a graph in order to compare the two methods.
+
+
+nb_bins = 100
+
+plt.hist(
+    y_cdf_1[0], cumulative=True, density=True, bins=nb_bins,
+    label='Absolute Residual Score', histtype='step', alpha=0.8
+)
+plt.hist(
+    y_cdf_2[0], cumulative=True, density=True, bins=nb_bins,
+    label='Normalized Residual Score', histtype='step', alpha=0.8
+)
+plt.vlines(
```

Comment:
We don't see the prediction (because they are the same), but then it's more confusing that anything else...

Reply:
I propose to merge the model predictions to avoid confusion.